nilearn/decoding/decoder.py

```diff
@@ -133,24 +138,36 @@ def _check_estimator(estimator):
 
 
 def _parallel_fit(estimator, X, y, train, test, param_grid, is_classification,
-                  scorer, mask_img, class_index, screening_percentile=100):
+                  scorer, mask_img, class_index, screening_percentile,
+                  do_clustering, clustering_percentile):
     """Find the best estimator for a fold within a job.
     This function tries several parameters for the estimator for the train and
     test fold provided and save the ones that performs best. These models are
     used afterwards to build the averaged model.
-
+    This tries may be performed after one step of clustering (ReNA) and one
+    step of feature screening.
     """
-    n_features = X.shape[1]
-    # Checking if the size of the mask image allows us to perform feature
-    # screening
-    selector = check_feature_screening(screening_percentile, mask_img,
-                                       is_classification)
-    do_screening = (n_features > 100) and selector is not None
-
     X_train, y_train = X[train], y[train]
     X_test, y_test = X[test], y[test]
 
-    if (selector is not None) and do_screening:
+    # for fREM Classifier and Regressor : start by doing a quick ReNA clustering
+    # to reduce the number of feature by agglomerating similar ones
+
+    if do_clustering and (clustering_percentile < 100):
+        n_clusters = int(X_train.shape[1] * clustering_percentile / 100.)
+        clustering = ReNA(mask_img, n_clusters=n_clusters, n_iter=20,
+                          threshold=1e-7, scaling=False)
```

Comment:
Is there any specific reason to hardcoded `n_iter`, `threshold` and `scaling` inside `_parallel_fit`? For example users might want to customize these parameters.

Reply:
When designing the PR with Gael we agreed on those:
- n_iter is enough for cases where clustering_percentile is very low but anyway it will be bypassed when the algorithm reaches the desired number of clusters.
- threshold is ok in general
- scaling I think it's a result of Andres thesis that it's not necessary.

Also you don't want to clutter a complex high-level object with technical parameters when those can be set right directly for all situations.